lib/spack/spack/database.py

```diff
@@ -339,7 +361,161 @@ def read_transaction(self):
         """Get a read lock context manager for use in a `with` block."""
         return ReadTransaction(self.lock, acquire=self._read)
 
-    def prefix_lock(self, spec):
+    def _failed_spec_path(self, spec):
+        """Return the path to the spec's failure file, which may not exist."""
+        if not spec.concrete:
+            raise ValueError('Concrete spec required for failure path for {0}'
+                             .format(spec.name))
+
+        return os.path.join(self._failure_dir,
+                            '{0}-{1}'.format(spec.name, spec.full_hash()))
+
+    def clear_failure(self, spec, force=False):
+        """
+        Remove any persistent and cached failure tracking for the spec.
+
+        Coordination between concurrent builds depends in large part on
+        prefix failure locks so the default is to not perform this operation
+        if there is a prefix failure lock, which indicates another process
+        may have taken out the lock.
+
+        The ability to programmatically detect and automatically retry a
```

Comment:
Make this a TODO

Reply:
This method provides that ability so is the TODO for making it available with the `spack` command?

Reply:
Should it be at the top of the module or here?

Reply:
Here’s fine